Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs

```diff
@@ -320,6 +320,74 @@ public void EmbeddedObject_WhenAdditiveExplicit_ShouldThrow()
             Assert.Throws<RealmSchemaValidationException>(() => Realm.GetInstance(conf), $"Embedded object {nameof(EmbeddedLevel3)} is unreachable by any link path from top level objects");
         }
 
+        [Test]
+        public void WriteCopy_DoesNotRedownloadData([Values(true, false)] bool originalEncrypted,
+                                                                 [Values(true, false)] bool copyEncrypted)
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var originalPartition = Guid.NewGuid().ToString();
+                var originalConfig = await GetIntegrationConfigAsync(originalPartition);
+                if (originalEncrypted)
+                {
+                    originalConfig.EncryptionKey = TestHelpers.GetEncryptionKey(42);
+                }
+
+                var copiedPartition = Guid.NewGuid().ToString();
```

Comment:
The copy should be synchronizing on the same partition - we can't change the partitions, so if this test passes, let's add a check `WriteCopy` to make sure the partition is the same.

Reply:
90d3d8e adds a test to check for that behaviour and also adjusts the other tests to use the same partition.